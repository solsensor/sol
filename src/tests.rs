use rocket::{http::Status, local::Client};
use std::fs;
use insta::assert_snapshot_matches;

fn setup() -> Client {
    fs::remove_file("sol.sqlite");
    super::db::run_migrations();
    let rocket = super::rocket();
    Client::new(rocket).expect("created test client")
}

#[test]
fn test_simple_ok() {
    let client = setup();
    let res = client.get("/").dispatch();
    assert_eq!(res.status(), Status::Ok);
}

#[test]
fn test_simple_not_ok() {
    let client = setup();
    let res = client.get("/invalid_endpoint").dispatch();
    assert_eq!(res.status(), Status::NotFound);
}

macro_rules! snap {
    (id: $id:ident, method: $method:ident, path: $path:expr,) => (
        #[test]
        fn $id() {
            let client = setup();
            let mut res = client.$method($path).dispatch();
            let contents = res.body_string().expect("no body content");
            assert_eq!(res.status(), Status::Ok);
            assert_snapshot_matches!(stringify!($id), contents);
        }
    )
}

snap!{
    id: get_users,
    method: get,
    path: "/api/users/all",
}
