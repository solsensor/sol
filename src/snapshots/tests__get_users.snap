---
created: "2019-07-09T04:13:35.593831572Z"
creator: insta@0.8.1
source: src/tests.rs
expression: contents
---
{"data":{"users":[]},"message":"found all users","status":"success"}
